src/testUtils.ts

```diff
@@ -106,67 +111,77 @@ function hasTestFunctionPrefix(name: string): boolean {
  * @param goConfig Configuration for the Go extension.
  */
 export function goTest(testconfig: TestConfig): Thenable<boolean> {
-	return new Promise<boolean>((resolve, reject) => {
-		outputChannel.clear();
-		if (!testconfig.background) {
-
-			outputChannel.show(true);
-		}
+	return cancelRunningTest().then(success => {
```

Comment:
Things to consider. FYI currently more than one test can be run all together.

When the test on save hook enabled `{"go.testOnSave": true}`, then every time user save file, then it run the test. When one test takes to long then it make sense if it canceled only when the second test run for the same package. If we save file belongs to different package, then it doesn't make sense. This can be happen when we hit "Save All" then multiple test will run.

Reply:
I hadn't considered that. However, I'm not sure how we can have multiple running tests and also have the ability to cancel running tests. Perhaps keeping a list of all processes in flight and then iterating to cancel all of them?

I've never intentionally run multiple test instances in VSCode before, so the test-per-package-on-save never occurred to me. It seems the output would get very jumbled, though.

@ramya-rao-a I'd love your thoughts here.

Reply:
@ramya-rao-a bump

Reply:
am here am here... 
Am prepping for an update at the moment, will get right back to this PR later this evening.

Thanks for your patience and sorry for the wait :(

Reply:
No worries, of course! Thank you for all that you do for the community! :)

Reply:
> Perhaps keeping a list of all processes in flight and then iterating to cancel all of them?

This won't help in the "save all" scenario if you intend to cancel running tests before running a new test. Even though to the user, save all seems like a single event, the extension gets notified once for each document being saved which then triggers the test on the corresponding package.

How about having the cancel test as a command (just like you have now), but have it triggered only by the user and not before running a new test?

You will still have to keep track of all running tests and cancel each of them

Reply:
That sounds like the best solution if we want to keep the functionality of running tests from multiple packages at once. I'll take a look at it when I have some time. Thanks!

Reply:
@tylerb @ramya-rao-a Probably there is a hack.
Extension react to the document save event. If we can delay this reaction for example for 1 second interval, capture all the events, after 1 second if we see more than one save events so run the test as one test session (run test only for affected packages).

Probably doesn't need to be 1sec, 500ms is enough.

Reply:
Something like this https://gist.github.com/uudashr/8d1d74e30b815c3a0eec634a737fdb9b

Reply:
For now, lets stick to the solution of the user cancelling the tests.

Save All triggering multiple processes also affects build/lint/vet. Let's tackle that in a separate PR